pkg/plugins/backendplugin/manager/plugin_settings_test.go

```diff
@@ -41,5 +42,55 @@ func TestPluginSettings(t *testing.T) {
 			require.Len(t, env, 3)
 			require.EqualValues(t, []string{"GF_PLUGIN_KEY1=value1", "GF_PLUGIN_KEY2=value2", "GF_VERSION=6.7.0"}, env)
 		})
+		t.Run("Should override config variable with environmental variable ", func(t *testing.T) {
+			_ = os.Setenv("GF_PLUGIN_KEY1", "sth")
+			ps := getPluginSettings("plugin", cfg)
+			env := ps.ToEnv("GF_PLUGIN", []string{"GF_VERSION=6.7.0"})
+			sort.Strings(env)
+			require.Len(t, env, 3)
+			require.EqualValues(t, []string{"GF_PLUGIN_KEY1=sth", "GF_PLUGIN_KEY2=value2", "GF_VERSION=6.7.0"}, env)
+
+			defer func() {
```

Comment:
Can you use `t.Cleanup` instead and do it straight after setting the environment variables? When you call `defer` here, it won't execute if the `require`s fail.